api/py/ai/chronon/repo/validator.py

```diff
@@ -284,7 +341,12 @@ def _validate_join(self, join: Join) -> List[str]:
         # Only validate the join derivation when the underlying groupBy is valid
         group_by_correct = all(not errors for errors in group_by_errors)
         if join.derivations and group_by_correct:
-            columns = set(get_pre_derived_join_columns(join))
+            columns = get_pre_derived_join_columns(join)
```

Comment:
nit: put this in else condition to avoid repeated calculation, because most joins are online

Reply:
updated